lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -807,20 +808,19 @@ def _set_scrape_metadata(self, event):
                 forward scrape jobs, alert rules, metrics endpoint host addresses
                 and related metadata to the Prometheus charm.
         """
-        event.relation.data[self._charm.unit]["prometheus_scrape_host"] = str(
-            self._charm.model.get_binding(event.relation).network.bind_address
-        )
+        self._set_unit_ip(event)
 
         if not self._charm.unit.is_leader():
             return
 
-        event.relation.data[self._charm.app]["scrape_metadata"] = json.dumps(self._scrape_metadata)
-        event.relation.data[self._charm.app]["scrape_jobs"] = json.dumps(self._scrape_jobs)
+        for relation in self._charm.model.relations[self._relation_name]:
+            relation.data[self._charm.app]["scrape_metadata"] = json.dumps(self._scrape_metadata)
+            relation.data[self._charm.app]["scrape_jobs"] = json.dumps(self._scrape_jobs)
 
-        if alert_groups := self._labeled_alert_groups:
-            event.relation.data[self._charm.app]["alert_rules"] = json.dumps(
-                {"groups": alert_groups}
-            )
+            if alert_groups := self._labeled_alert_groups:
+                relation.data[self._charm.app]["alert_rules"] = json.dumps(
```

Comment:
Isn't app data shared across all relation's data?

Reply:
No I do not think so. As far as I understand there is an app data bucked per relation.

Reply:
This may not be the case in the following scenario which I observed multiple times with alertmanager:
1. Deploy number of units from the start, such that the leader is not unit/0.
2. Scale down a number of units under the current leader.
3. Units are removed but the new leader is elected only ~10 seconds after scaledown.